compiler/GHC/Unit/State.hs

```diff
@@ -1464,9 +1478,10 @@ validateDatabase cfg pkg_map1 =
 
 mkUnitState
     :: Logger
+    -> DynFlags
     -> UnitConfig
     -> IO (UnitState,[UnitDatabase UnitId])
-mkUnitState logger cfg = do
+mkUnitState logger dflags cfg = do
```

Comment:
No no no. `mkUnitState` must only be created from `UnitConfig`. Put whatever you need from DynFlags into it.